### ISSUE TITLE
MaaS: Inconsistent version of psutil being installed

### ISSUE THREAD

**Issue by mkam** [rcbops/rpc-openstack#1984]
##### What type of issue is this? Please tick the appropriate box.
- [x] bug - existing functionality that does not work for you as documented
- [ ] feature - new functionality that you would like to see added

##### What is the expected behavior?
All nodes and containers should have psutil==1.2.1 installed. Subsequently, all process status alarms (filebeat_process_status, rsyslogd_process_status) should be OK.

##### What is actually happening?
In the multi-node AIO environments, some containers have psutil==1.2.1 while others have the psutil==5.1.3, which was the latest release at the time. The process alarms on the containers with the newer versions go to CRITICAL with the following issue:
```
Traceback (most recent call last):
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 123, in <module>
    main(args)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 113, in main
    process_names=args.processes)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 67, in check_process_running
    procs = get_processes()
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 49, in get_processes
    if venv_wrapper_check():
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 32, in venv_wrapper_check
    if 'run_plugin_in_venv.sh' in ' '.join(parent_process.cmdline):
AttributeError: 'function' object has no attribute 'cmdline'
```

##### What were you attempting to do when you encountered this error (give as much detail as possible)? 
Created a multi-node AIO server, ran setup-maas.yml, ran verify-maas.yml.

##### How can we reproduce this issue?
Deploy a multi-node AIO environment manually or use https://rpc.jenkins.cit.rackspace.net/job/OnMetal_Multi_Node_AIO/

##### What is your evaluation on the importance of this issue, and why?

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [x] master
- [x] newton-14.0
- [ ] mitaka-13.1
- [ ] liberty-12.2

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [ ] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [ ] This change includes additional testing
- [ ] This change is tested by the existing setup

Additional testing links:

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [ ] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by xeregin** [comment]
This applies to the LEM labs as well.

**Comment by major** [comment]
See #1985

**Comment by melissa-kam** [comment]
The no attribute 'cmdline' error is gone, but now in container process checks, there is a new AttributeError for psutil 5.x.

```
(maas-r13.1.0rc1) root@logging1:/etc/rackspace-monitoring-agent.conf.d# pip freeze | grep psutil
psutil==5.2.0

(maas-r13.1.0rc1) root@logging1:/etc/rackspace-monitoring-agent.conf.d# /usr/lib/rackspace-monitoring-agent/plugins/process_check.py -c logging1_elasticsearch_container-3169de90 java
status error parent_pid=init_pid)\n  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 64, in get_processes\n    procs = [x for x in psutil.Process(parent_pid).get_children()]\nAttributeError: 'Process' object has no attribute 'get_children'\n
Traceback (most recent call last):
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 137, in <module>
    main(args)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 127, in main
    process_names=args.processes)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 88, in check_process_running
    procs = get_processes(parent_pid=init_pid)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 64, in get_processes
    procs = [x for x in psutil.Process(parent_pid).get_children()]
AttributeError: 'Process' object has no attribute 'get_children'
```
```
(maas-r13.1.0rc1) root@logging1:/etc/rackspace-monitoring-agent.conf.d# pip freeze | grep psutil
psutil==5.1.3

(maas-r13.1.0rc1) root@logging1:/etc/rackspace-monitoring-agent.conf.d# /usr/lib/rackspace-monitoring-agent/plugins/process_check.py -c logging1_elasticsearch_container-3169de90 java
status error parent_pid=init_pid)\n  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 64, in get_processes\n    procs = [x for x in psutil.Process(parent_pid).get_children()]\nAttributeError: 'Process' object has no attribute 'get_children'\n
Traceback (most recent call last):
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 137, in <module>
    main(args)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 127, in main
    process_names=args.processes)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 88, in check_process_running
    procs = get_processes(parent_pid=init_pid)
  File "/usr/lib/rackspace-monitoring-agent/plugins/process_check.py", line 64, in get_processes
    procs = [x for x in psutil.Process(parent_pid).get_children()]
AttributeError: 'Process' object has no attribute 'get_children'
```

The ones that don't check containers are passing.
```
(maas-r13.1.0rc1) root@logging1:/etc/rackspace-monitoring-agent.conf.d# pip freeze | grep psutil
psutil==5.2.0

(maas-r13.1.0rc1) root@logging1:/etc/rackspace-monitoring-agent.conf.d# /usr/lib/rackspace-monitoring-agent/plugins/process_check.py filebeat
status okay
metric filebeat_process_status uint32 1
```

**Comment by mancdaz** [comment]
@major @alextricity25 does the patch from #1993 now need to be backported to the newton-14.0 branch?

**Comment by mancdaz** [comment]
Answering my own question. Yes it does, but https://github.com/rcbops/rpc-openstack/pull/1997 needs to merge first so the cherry-pick will apply cleanly

**Comment by major** [comment]
You're right, @mancdaz. (I miss Gerrit.) :(
